Helper/Order.php

```diff
@@ -712,12 +726,29 @@ public function saveUpdateOrder($reference, $storeId = null, $boltTraceId = null
                     if ($transaction->status == Payment::TRANSACTION_AUTHORIZED) {
                         $this->voidTransactionOnBolt($transaction->id, $storeId);
                     }
-                    $this->bugsnag->notifyException($exception);
-                    return $this;
+
+                ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                //// Allow missing quote failed hooks to resend 10 times so the Administrator can be notified via email
+                ///  On the eleventh time that the failed hook was sent to Magento, we return $this in order to have the hook return successfully.
+                ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+                    /** @var \Bolt\Boltpay\Model\Log $logFactory */
+                    $logFactory = $this->logFactory->create();
+                    /** @var \Bolt\Boltpay\Model\ResourceModel\Log\Collection $logCollectionFactory */
+                    $logCollectionFactory = $this->logCollectionFactory->create();
+                    if ($log = $logCollectionFactory->getLogByTransactionId($transaction->id, $hookType)) {
+                        $numberOfMissingQuoteFailedHooks = $log->getNumberOfMissingQuoteFailedHooks();
+                        if ($numberOfMissingQuoteFailedHooks > 10) {
+                            return $this;
+                        }
+
+                        $logFactory->updateLog($log->getId(), $numberOfMissingQuoteFailedHooks);
```

Comment:
lets call the method `incrementAttemptCount`
```suggestion
                        $logFactory->incrementAttemptCount($log->getId(), $numberOfMissingQuoteFailedHooks);
```